route/v1/types.go

```diff
@@ -139,8 +139,90 @@ type RouteSpec struct {
 	// +kubebuilder:validation:Enum=None;Subdomain;""
 	// +kubebuilder:default=None
 	WildcardPolicy WildcardPolicyType `json:"wildcardPolicy,omitempty" protobuf:"bytes,7,opt,name=wildcardPolicy"`
+
+	// httpHeaders defines policy for HTTP headers.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	HTTPHeaders *RouteHTTPHeaders `json:"httpHeaders,omitempty" protobuf:"bytes,9,opt,name=httpHeaders"`
+}
+
+// RouteHTTPHeaders defines policy for HTTP headers.
+type RouteHTTPHeaders struct {
+	// actions specifies options for performing action on headers.
+	// For Setting Headers one can provide a value for the header whereas for deleting one needs to provide header name.
+	// +optional
+	Actions RouteHTTPHeaderActions `json:"actions,omitempty" protobuf:"bytes,1,opt,name=actions"`
+}
+
+// RouteHTTPHeaderActions defines whether we want to set Response or Request Headers.
+type RouteHTTPHeaderActions struct {
+	// Response is a list of http response headers which either need to set or deleted.
+	Response []RouteHTTPHeader `json:"response,omitempty" protobuf:"bytes,1,opt,name=response"`
+	// Request is a list of http request headers which either need to set or deleted.
+	Request []RouteHTTPHeader `json:"request,omitempty" protobuf:"bytes,2,opt,name=request"`
+}
+
+// RouteHTTPHeader specifies options for set/delete a header.
+type RouteHTTPHeader struct {
+	// name specifies a header name to be set or deleted.  Its value must be a valid HTTP header.
+	// name as defined in RFC 2616 section 4.2.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// value specifies fields to mention the header details and the action you want to perform on them.
+	Value RouteHTTPHeaderActionUnion `json:"value" protobuf:"bytes,2,opt,name=value"`
```

Comment:
As I mentioned on the IngressController API changes, I believe this should be inlined or renamed.

Reply:
hi @Miciah ! I had discussed this with @JoelSpeed and he asked me to do so.

Reply:
Think we covered this somewhere else, but to close off this thread, inline unions have proven to be confusing for users so API conventions currently recommend against them within OpenShift at least. They are still permitted upstream though